crates/oxc_parser/src/lexer/number.rs

```diff
@@ -212,10 +260,48 @@ mod test {
         }
     }
 
+    // octal
+    static_assertions::const_assert_eq!(octal_byte_to_value(b'0'), 0);
+    static_assertions::const_assert_eq!(octal_byte_to_value(b'7'), 7);
+
+    // hex
+    static_assertions::const_assert_eq!(hex_byte_to_value(b'0'), 0);
+    static_assertions::const_assert_eq!(hex_byte_to_value(b'9'), 9);
+    static_assertions::const_assert_eq!(hex_byte_to_value(b'A'), 10);
+    static_assertions::const_assert_eq!(hex_byte_to_value(b'F'), 15);
+    static_assertions::const_assert_eq!(hex_byte_to_value(b'a'), 10);
+    static_assertions::const_assert_eq!(hex_byte_to_value(b'f'), 15);
+
     #[test]
-    #[allow(clippy::excessive_precision)]
+    #[allow(clippy::excessive_precision, clippy::cast_precision_loss)]
     fn test_int_precision() {
         assert_eq!(parse_int("9007199254740991", Kind::Decimal, false), Ok(9007199254740991.0));
+        assert_eq!(
+            parse_int("0x10000000000000000", Kind::Hex, false),
+            Ok(0x10000000000000000_i128 as f64)
+        );
+        assert_eq!(
+            parse_int("0o40000000000000000", Kind::Octal, false),
+            Ok(0o40000000000000000_i128 as f64)
+        );
+        assert_eq!(
+            parse_int(
+                "0b0001000000000000000000000000000000000000000000000000000000000000",
```

Comment:
I don't think this test is covering the case that we need it to. You have 60 zeros after the 1 here. If you add 4 more, I think you'll find the test fails. `parse_binary` will need changing in same manner as `parse_octal` and `parse_hex` to fix that.

Reply:
Good catch, fixed